src/infrastructure/transaction/CreateTransactionFromDTO.ts

```diff
@@ -200,6 +201,21 @@ const CreateStandaloneTransactionFromDTO = (transactionDTO, transactionInfo, isE
                 : undefined,
             transactionInfo,
         ).setPayloadSize(transactionDTO.size);
+    } else if (type === TransactionType.MOSAIC_SUPPLY_REVOCATION) {
+        const dto = transactionDTO as MosaicSupplyRevocationTransactionDTO;
+        return new MosaicSupplyRevocationTransaction(
+            (dto.network as any) as NetworkType,
```

Comment:
why isn't `transactionDTO.network` sufficient as it is for all of the other transactions?

Reply:
The other mappers don't cast to the generated open api DTO, they use "any" object. I would prefer to cast to the DTO so we are actually type-safe and we can validate we didn't miss any field openapi. The question would be if we want to improve all other mappers or not. I could do a clean-up in this SDK but I wanted to keep code changes to the minimum.

About the network field, `dto.network` is of type `NetworkTypeEnum` different enum than `NetworkType` (although they have the same values)

We are moving away from the model wrappers on the new SDKs. Although the wrappers are a bit nicer for the user, it adds a level of complexity that is expensive, especially if creating SDKs for multiple languages. For example, we would need this PR if we use catbuffer and OpenAPI generated libraries directly.

Reply:
Update, I'm removing the generated DTO downcast to be consistent.